### DISCUSSION TITLE
Foundry-FundMe. An error after remmapings.

### DISCUSSION THREAD

**Discussion by genkisudo** [Cyfrin/foundry-full-course-cu#1381] [Q&A] [answered]
Hi, I'm getting the error after remapping the foundry.toml file. 
Screenshot attached. Thanks. 

I installed 0.6.1 chainlink contracts version, wonder if I should have installed 0.8.0

<img width="1228" alt="Screenshot 2024-02-05 at 11 26 08" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/57138960/01691df1-aa80-41c0-9841-9fe5f1fa8f7d">

**Comment by heheboii11** [comment, marked as answer]
Had the same problem I cannot find the exact solution but import directly instead of using the remappings

 import {VRFCoordinatorV2Interface} from "lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";

**Reply by genkisudo** [reply]
Thanks so much, it worked for me as well!

**Reply by DavidWilliamsQA** [reply]
you can use the mapping but you need to remove the 'shared' part of the path so it should be 
```
"@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"
```

instead of 
```
"@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"
```

when importing

**Reply by Eshan-Sharma** [reply]
This solution worked for my error as well,
Error
```
Compiler run failed:
Error (9553): Invalid type for argument in function call. Invalid implicit conversion from contract AggregatorV3Interface to contract AggregatorV3Interface requested.
  --> src/FundMe.sol:32:45:
   |
32 |         require(msg.value.getConversionRate(s_pricefeed) >= MINIMUM_USD, "You need to spend more ETH!");
   |                                             ^^^^^^^^^^^
 ```

**Comment by saint-gmoney0** [comment]
hi everyone am new to web3 development hope am welcome
@PatrickAlphaC

**Reply by arunprabu563** [reply]
![IMG_20240317_165829.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/85756382/7d1e8260-effe-43b1-bc2f-4700b39759ae)

After this import, i got an error like this price converter , help me

**Reply by arunprabu563** [reply]
@heheboii11 can you help?

**Reply by AndroKasrashvili** [reply]
add "../" before "lib/..." in your import

**Reply by arunprabu563** [reply]
still the same problem,

**Reply by arunprabu563** [reply]
Help me

**Reply by heheboii11** [reply]
send me full code here with proper formatting and also mention where are getting the error

**Reply by heheboii11** [reply]
Ig you did not import ```AggregatorV3Interface``` in ```PriceConverter.sol```, please check this and import this you error shoukd be rectified. If you are still facing the same issue then please send the full code of your ```PriceConverter.sol```

**Reply by Sharvil1123** [reply]
use this import
```
 {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
 ```

**Reply by arunprabu563** [reply]
/ SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";


// Why is this a library and not abstract?
// Why not an interface?
library PriceConverter {
    // We could make this public, but then we'd have to deploy it
    function getPrice() internal view returns (uint256) {
        // Sepolia ETH / USD Address
        // https://docs.chain.link/data-feeds/price-feeds/addresses
        AggregatorV3Interface priceFeed = AggregatorV3Interface(
            0x694AA1769357215DE4FAC081bf1f309aDC325306
        );
        (, int256 answer, , , ) = priceFeed.latestRoundData();
        // ETH/USD rate in 18 digit
        return uint256(answer * 10000000000);
    }

    // 1000000000
    function getConversionRate(
        uint256 ethAmount
    ) internal view returns (uint256) {
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        // the actual ETH/USD conversion rate, after adjusting the extra 0s.
        return ethAmountInUsd;
    }
}

my price converter code

**Reply by heheboii11** [reply]
did you install chainlink contract kit ?, If not first try to install them by using ```forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit```, if you have already installed them and if you are facing error then try to add remappings correctly

**Comment by Sharvil1123** [comment]
Hey @kamikagome check the import location of AggregatorV3Interface given as - 
```
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
 ```
Had the same issue, check the location of the function and found the answer. Just removed the "/shared" from the import location.

**Reply by arunprabu563** [reply]
i got it , thanks for your help,
